source/material_model/interface.cc

```diff
@@ -914,6 +914,32 @@ namespace aspect
     }
 
 
+    template<int dim>
+    PhasePropertyOutputs<dim>::PhasePropertyOutputs (const unsigned int n_points,
+                                                     const std::vector<std::string> property_names)
+      :
+      NamedAdditionalMaterialOutputs<dim>(property_names),
+      phase_property_values(n_points, std::vector<double>(property_names.size(), numbers::signaling_nan<double>()))
+    {}
+
+
+
+    template<int dim>
+    std::vector<double>
+    PhasePropertyOutputs<dim>::get_nth_output(const unsigned int idx) const
+    {
+      // we have to extract the nth phase property output
+      // but the vector in the material model outputs is sorted so that the
+      // number of evaluation points (and not the number of phase properties)
+      // is the outer vector
```

Comment:
At the moment, the Steinberger model loops over each evaluation point, so it was easier to pass in.temperature[i], in.pressure[i], in.composition[i] ... to the new phase_volume_fraction function. 

But I could equally pass the function a reference to `in` and do both the inner and outer loops within the function. This will mean a few more lines of code in Steinberger, but interface.cc will be neater (and a bit faster).

Reply:
Now implemented the other way round :)